File: nodejs/expressmvc/prescricao.js
const fs = require('fs');
const path = require('path');
const { StorageMock } = require('./storage-mock');
const uuidv4 = require('uuid/v4');
const PDFDocument = require('pdfkit');

const APP_ROOT = process.cwd();

class Prescricao {
    static generateReceitaSimples(name, crm, crmUf){

        const defaultLabel = [ 
            'Nome do(a) Médico(a):',
            'CRM:',
            'CRM UF:',
            'Nome do Paciente:', 
            'Local de Atendimento:', 
            'Telefone:', 
            'Endereço:', 
            'Cidade:', 
            'Bairro:', 
            'CNES:', 
            'UF:',
            'Data de Emissão:', 
            'Prescrição:',
        ];
        const defaultValues = [
            name,
            crm,
            crmUf,
            'Maria da Silva',
            'Clínica Local',
            '+00 (00) 0000-0000',
            'Complexo Hospitalar',
            'Brasília',
            'Bairro do Mar',
            '0000000',
            'DF',
            '00/00/0000',
            'Dipirona ------- 1 comprimido de 12 em 12 horas por 3 dias',
        ];

        // Create temporary file.
        const outputFile = `${uuidv4()}.pdf`;
        StorageMock.createAppDataSync(); // Make sure the 'app-data' folder exists.
        let dest = path.join(APP_ROOT, 'app-data', outputFile);

        const doc = new PDFDocument({
            size: 'A4',
            // REQUIRED!
            // Add all the following metadata
            info: {
                '2.16.76.1.12.1.1': '',     // Prescrição de medicamento
                '2.16.76.1.4.2.2.1': crm,   // CRM
                '2.16.76.1.4.2.2.2': crmUf, // CRM UF
                '2.16.76.1.4.2.2.3': '',    // Especialidade
                '2.16.76.1.4.2.3.1': '',    // CRF
                '2.16.76.1.4.2.3.2': '',    // CRF UF
                '2.16.76.1.4.2.3.3': '',    // Especialidade
              }
            });

        // Stream contents to dest file
        doc.pipe(fs.createWriteStream(dest));

        // Add title.
        doc.fontSize(20);
        doc.font('Times-Bold')
            .text('RECEITUÁRIO SIMPLES', {align: 'center'})
            .moveDown(1);

        // Initialize Form
        doc.fontSize(12).font('Helvetica');

        // Add text
        defaultLabel.forEach((label, idx) => {
            doc.text(label, 72)
            .moveUp(1)
            .text(defaultValues[idx], 210)
            .moveDown(1);
        });

        doc.end();

        return outputFile;
    }
}

exports.Prescricao = Prescricao;

File: python/flask/sample/prescricao.py
import binascii
import os
import re
import uuid
from fpdf import FPDF

"""
    This function generates a new PDF document with all the main information for a medical prescription.
    It uses the package fpdf to create and edit the PDF, although there are different python packages 
    eith this purpose.

"""
def generate_receita_simples(name, crm, crmUf):
    # create new PDF document in A4 portrait and the measure is in milimeter
    document = FPDF('P', 'mm', 'A4');
    # create a new page 
    document.add_page()
    #left, top, right margins
    document.set_margins(1,2,1)
    # font: ariel, regular font
    document.set_font('Arial', "B", 16)

    # REQUIRED!!!
    # set all the PDF metadata
    # when value is not specified, use empty strings
    document.set_title("RECEITUÁRIO SIMPLES")
    document.set_subject("CRM: "+crm+ " UF: "+crmUf)
    especialidade = ""
    crf = ""
    crfUf = ""
    document.set_keywords("Especialidade: " + especialidade + " CRF: "
        + crf + "CRF-UF: " + crfUf)
    document.set_author(name)

    # set main information
    document.ln(8)
    document.cell(200,1,"NOME DO(A) MÉDICO(A): " + name,0,1,'L')
    document.ln(8)
    document.cell(200,1,"CRM: " + crm,0,1,'L')
    document.ln(8)
    document.cell(200,1,"UF: " + crmUf,0,1,'L')
    document.ln(12)

    
    # set remain content
    default_info = {"Telefone" :            "+00 (00) 0000-0000",
                   "Endereço" :             "Complexo Hospitalar", 
                    "Bairro" :              "Bairro do Mar",
                    "Cidade" :              "Brasília",
                    "UF" :                  "DF",
                    "Local de Atendimento" :"Clínica Local",
                    "CNES" :                "0000000",
                    "Nome do Paciente" :    "Maria da Silva",
                    "Prescrição" :          "Remédio X ---------- 1 comprimido de 12 em 12 horas por 3 dias",
                    "Data de Emissão" :     "00/00/0000"}

    document.set_font('Arial', "", 16)

    # associate each label to one value
    for label, value in default_info.items():
        document.cell(200, 1, label+": "+value,0, 2, 'L')
        document.ln(8)

    # Generate a unique filename.
    filename = '%s.%s' % (str(uuid.uuid4()), "pdf")
    
    # Return the document as a string, named with the generated filename
    return document.output(filename,'S').encode('latin-1'); 
